File: past-competitions/2018/src/brett-cpp/Prob10.c
#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(int argc, char *argv[])
{
	// open file for reading
	FILE *inputFile = fopen("Prob10.in.txt", "r");
	int testCases;

	// read number of test cases
	fscanf(inputFile, "%d\n", &testCases);
	
	int boardX, boardY, startX, startY, endX, endY;

	// execute test cases
	while(testCases > 0){
		// read in data
		fscanf(inputFile, "%d,%d\n", &boardX, &boardY);
		fscanf(inputFile, "%d,%d\n", &startX, &startY);
		fscanf(inputFile, "%d,%d\n", &endX, &endY);

		// size of board is not important; a bishop can access any space
		// that's the same color as the one it started on.
		// white spaces will be on "even" squares; add X and Y, if the
		// result is even, the space is white.
		if((startX + startY) % 2 == (endX + endY) % 2){
			printf("Yes\n");
		}
		else{
			printf("No\n");
		}

		testCases = testCases - 1;
	}

	// close the file
	fclose(inputFile);
}

File: past-competitions/2018/src/brett-python/Prob10.py
# Open the input file
with open("Prob10.in.txt", "rt") as inputFile:
    # Read the number of test cases (trim out the newline)
    cases = int(inputFile.readline().replace("\n", ""))
    # For each test case
    for caseNum in range(cases):
        # Read the data
        boardsize = inputFile.readline().replace("\n", "")
        start = inputFile.readline().replace("\n", "")
        end = inputFile.readline().replace("\n", "")
        # board size is irrelevant
        # each space can be "even" or "odd"
        # add its coordinates; if the result is even, the space is "even"
        # a bishop starting on an "even" space can only move to "even" spaces
        # split the coordinates
        startCoords = start.split(",")
        endCoords = end.split(",")
        # add the coordinates
        startValue = int(startCoords[0]) + int(startCoords[1])
        endValue = int(endCoords[0]) + int(endCoords[1])
        # compare evenness
        if startValue % 2 == endValue % 2:
            print("Yes")
        else:
            print("No")
        
